File: Aplicaciones Distribuidas/PARCIAL2/HILOS_DOTNET_GRUPO05/TALLER_HILOS_PRACTICA3/APLICATIVO/175_TALLER_HILOS_GRUPO05/Controlador/BancoControlador.cs
using _175_TALLER_HILOS_GRUPO05.Modelo;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace _175_TALLER_HILOS_GRUPO05.Controlador
{
    public class BancoControlador
    {
        public void acreditarSaldoCuentas(BancoModelo banco, double saldo)
        {
            double[] cuentasAcreditadas = new double[100];
            for (int i = 0; i < cuentasAcreditadas.Length; i++)
            {
                cuentasAcreditadas[i] = saldo;
            }
            banco.cuentas = cuentasAcreditadas;
        }

        public void transferirEntreCuentas(BancoModelo banco, int cuentaOrigen, int cuentaDestino, double cantidad)
        {
            if (banco.cuentas[cuentaOrigen] < cantidad)
            {
                return;
            }

            Console.WriteLine(Thread.CurrentThread.ManagedThreadId);
            banco.cuentas[cuentaOrigen] -= cantidad;
            Console.WriteLine(cantidad.ToString("0.##") + " de " + cuentaOrigen + " para " + cuentaDestino);
            banco.cuentas[cuentaDestino] += cantidad;
            Console.WriteLine("Saldo Total: " + obtenerSaldoTotal(banco));
        }

        public double obtenerSaldoTotal(BancoModelo banco)
        {
            double saldoTotal = 0;
            foreach (double saldo in banco.cuentas)
            {
                saldoTotal += saldo;
            }
            return saldoTotal;
        }
    }
}

File: Aplicaciones Distribuidas/PARCIAL1/05. HILOS_NETBEANS_CHINLLE_CHUQUIMARCA_COBACANGO/TALLER_HILOS_PRACTICA3/Aplicativo/174_TALLER_HILOS_GRUPO05/src/ec/edu/monster/controlador/BancoControlador.java
package ec.edu.monster.controlador;

import ec.edu.monster.modelo.BancoModelo;

public class BancoControlador {
    public void acreditarSaldoCuentas(BancoModelo banco, double saldo) {
        double[] cuentasAcreditadas = new double[100];
        for (int i = 0; i < cuentasAcreditadas.length; i++) {
            cuentasAcreditadas[i] = saldo;
        }

        banco.cuentas = cuentasAcreditadas;
    }

    public void transferirEntreCuentas(BancoModelo banco, int cuentaOrigen, int cuentaDestino, double cantidad) {
        if (banco.cuentas[cuentaOrigen] < cantidad) {
            return;
        }

        System.out.println(Thread.currentThread());

        banco.cuentas[cuentaOrigen] -= cantidad;

        System.out.printf("%10.2f de %d para %d\n", cantidad, cuentaOrigen, cuentaDestino);

        banco.cuentas[cuentaDestino] += cantidad;

        System.out.printf("Saldo Total: %10.2f\n", obtenerSaldoTotal(banco));
    }

    public double obtenerSaldoTotal(BancoModelo banco) {
        double saldoTotal = 0;

        for (double saldo : banco.cuentas) {
            saldoTotal += saldo;
        }

        return saldoTotal;
    }
}
